hw/dv/data/dsim/dsim.hjson

```diff
@@ -0,0 +1,122 @@
+
+  // Indicate the tool specific helper sources - these are copied over to the
+  // {tool_srcs_dir} before running the simulation.
+  tool_srcs:  ["{proj_root}/hw/dv/tools/dsim/*"]
+
+  build_opts: [
+               "-work {build_dir}/dsim_out",
+               "-genimage image",
+               "-sv",
+
+               // UVM
+               "+incdir+{UVM_HOME}/src",
+               "{UVM_HOME}/src/uvm_pkg.sv",
+               "+acc+rwb -sv_lib {DSIM_HOME}/lib/libuvm_dpi.so",
+
+               "-timescale 1ns/1ps",
+               "-f {sv_flist}",
+               "+incdir+{build_dir}",
+               // Suppress following DSim errors and warnings:
+               //   EnumMustBePositive - UVM 1.2 violates this (b/150231731)
+               "-suppress EnumMustBePositive",
+              ]
+
+  run_opts:   [
+               "-work {build_dir}/dsim_out",
+               "-image image",
+               // UVM DPI
+               "+acc+rwb -sv_lib {DSIM_HOME}/lib/libuvm_dpi.so",
+               "+ntb_random_seed={seed}",
+               "+UVM_TESTNAME={uvm_test}",
+               "+UVM_TEST_SEQ={uvm_test_seq}"]
+
+  // Coverage related.
+  cov_db_dir: "{scratch_path}/coverage/{build_mode}.vdb"
+
+  // Individual test specific coverage data - this will be deleted if the test fails
+  // so that coverage from failing tests is not included in the final report.
+  cov_db_test_dir_name: "{run_dir_name}.{seed}"
+  cov_db_test_dir: "{cov_db_dir}/snps/coverage/db/testdata/{cov_db_test_dir_name}"
+
+  // Merging coverage.
+  // "cov_db_dirs" is a special variable that appends all build directories in use.
+  // It is constructed by the tool itself.
+  cov_merge_dir:    "{scratch_base_path}/cov_merge"
+  cov_merge_db_dir: "{cov_merge_dir}/merged.vdb"
+  cov_merge_cmd:    "{job_prefix} urg"
+  cov_merge_opts:   [
+                    ]
+  // Generate coverage reports in text as well as html.
+  cov_report_dir:       "{scratch_base_path}/cov_report"
+  cov_report_cmd:       "{job_prefix} urg"
+  cov_report_opts:  [
+                    ]
+  cov_report_txt:       "{cov_report_dir}/dashboard.txt"
+  cov_report_page:      "cov_report/dashboard.html"
+
+  // Analyzing coverage - this is done by invoking --cov-analyze switch. It opens up the
+  // GUI for visual analysis.
+  cov_analyze_dir:  "{scratch_base_path}/cov_analyze"
+  cov_analyze_cmd:  "{job_prefix} verdi"
+  cov_analyze_opts: ["-cov",
+                     "-covdir {cov_merge_db_dir}",
+                     "-line nocasedef"
+                     "-elfile {vcs_cov_excl_files}"]
+
+  // Vars that need to exported to the env.
+  exports: [
+  ]
+
+  // Defaults for DSim
+  cov_metrics:          "line+cond+fsm+tgl+branch+assert"
+  vcs_cov_hier:         ""
+  vcs_cov_assert_hier:  ""
+  vcs_cov_excl_files:   []
+
+  // pass and fail patterns
+  build_fail_patterns: ["^Error-.*$"]
+  run_fail_patterns:   ["^Error-.*$"] // Null pointer error
+
+  wave_file: "{scratch_path}/dsim_wave.vcd"
```

Comment:
Shouldn't this be unique to each test run? IMO this should be `{run_dir}/waves.vcd`.